src/screens/TopicPageScreen/elements/TopicDomainHeader.js

```diff
@@ -11,31 +11,29 @@ import {COLORS} from '../../../utils/theme';
 
 const TopicDomainHeader = (props) => {
   const {domain, handleOnMemberPress, hideSeeMember, isFollow, memberCount} = props;
-  const isUserFollow = props?.initialData?.isFollowing ? props.initialData.isFollowing : isFollow;
 
   const handlePress = () => {
-    if (isUserFollow) {
+    if (isFollow) {
       handleOnMemberPress();
     } else {
       SimpleToast.show('Only community members can see other members', SimpleToast.SHORT);
     }
   };
+  const shouldDisplay = !(props.topicDetail?.cover_path?.length > 0)
+    ? false
+    : props.hasSearch
+    ? props.topicDetail?.cover_path?.length > 0
+    : props.isHeaderHide;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic for determining `shouldDisplay` seems overly complex and might be simplified. It's also unclear what each condition represents without comments or more descriptive variable names. Consider refactoring this code for clarity and simplicity.
```diff
-  const shouldDisplay = !(props.topicDetail?.cover_path?.length > 0)
-    ? false
-    : props.hasSearch
-    ? props.topicDetail?.cover_path?.length > 0
-    : props.isHeaderHide;
+  const hasCoverPath = props.topicDetail?.cover_path?.length > 0;
+  const shouldDisplay = hasCoverPath && (props.hasSearch || props.isHeaderHide);
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->